README.md

```diff
@@ -1,116 +1,170 @@
-from your_project_model import RetNetModel  # Replace with your actual model import
+```bash
+git clone https://github.com/DRAGNLabs/301r_retnet.git
+cd 301r_retnet
+```
+
+### 1: Development Environment
+
+Ensure you have Python 3.11 installed. If you do not have Python 3.11, you can download it from the [official Python website](https://www.python.org) or use a package manager.
+
+Optionally, you can create a virtual environment. Then install all the necessary dependencies. An example with [Mamba](https://mamba.readthedocs.io/en/latest/index.html) is:
+```bash
+# Optionally create a new Mamba environment with Python 3.11 and specify a name
+mamba create -n <YOUR_ENV_NAME> python=3.11
 
-# Basic hyperparameters
-hyperparams = {
-    "learning_rate": 0.001,
-    "batch_size": 64,
-    "num_layers": 4,
-    "embed_dim": 256,
-    "num_heads": 8,
-    "dropout_rate": 0.1
-}
+# Activate the Mamba environment
+mamba activate <YOUR_ENV_NAME>
+```
 
-model = RetNetModel(num_layers=hyperparams["num_layers"],
-                    embed_dim=hyperparams["embed_dim"],
-                    num_heads=hyperparams["num_heads"],
-                    dropout_rate=hyperparams["dropout_rate"])
-optimizer = torch.optim.Adam(model.parameters(), lr=hyperparams["learning_rate"])
+Once your environment has been prepared, install all required packages:
+```bash
+pip install -r requirements.txt
 ```
 
-#### 2. **Implementing Learning Rate Scheduler**:
+### 2: YAML Configuration Files
 
-An example of using a learning rate scheduler.
+This project uses YAML configuration files to store all pipeline parameters and paths. The design choice of the YAML file is intended to eliminate repetition of commonly used parameters across code, as well as simplify future changes and refactors, allow developers to add new parameters, and make all settings visible to the user in one consolidated place.
 
-```python
-scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=10, gamma=0.1)
+To prepare a YAML config file, copy [template_config.yaml](./configs/template_config.yaml) into the [user_configs](./configs/user_configs/) folder. Fill out all parameters accordingly. Absolute paths are preferred for any path variables, but the repository is set up to work flexibly with any desired directory structure.
 
-# Training loop
-for epoch in range(num_epochs):
-    # Training process
-    ...
-    scheduler.step()  # Update learning rate
-```
+*Note:* In most cases, YAML does not expect strings. Adding quotation marks around arguments in the config file can lead to unexpected errors.
 
-#### 3. **Random Search for Hyperparameter Tuning**:
+### 3: Script Execution
 
-Pseudo-code for implementing random search.
+Once a YAML config file is prepared, it can be passed into any script in the pipeline. Before you run any scripts, it is recommended to copy all of them into the [user_scripts](./scripts/user_scripts/) directory and modify the scripts to point to the right config file.
```

Comment:
Look at the raw text below how to implement this in GitHub. I think this point should be emphasized.

> [!TIP]
> Once a YAML config file is prepared, it can be passed into any script in the pipeline. Before you run any scripts, it is recommended to copy all of them into the [user_scripts](./scripts/user_scripts/) directory and modify the scripts to point to the right config file.

Reply:
I see this in the script, did you add it already?